backend/LeetWars.Builder/Services/XmlTestResultParserService.cs

```diff
@@ -0,0 +1,164 @@
+﻿using LeetWars.Builder.Interfaces;
+using LeetWars.Builder.Models;
+using System.Globalization;
+using System.Xml;
+
+namespace LeetWars.Builder.Services
+{
+    public class XmlTestResultParserService : IXmlTestResultParserService
+    {
+        public TestsOutput ParseCSharpTestResult(string xmlTestResult)
+        {
+            XmlDocument xmlDoc = new();
+
+            xmlDoc.LoadXml(xmlTestResult);
+
+            XmlNamespaceManager nsManager = new (xmlDoc.NameTable);
+
+            if (xmlDoc.DocumentElement?.NamespaceURI != null)
+            {
+                nsManager.AddNamespace("ns", xmlDoc.DocumentElement.NamespaceURI);
+            }
+
+            XmlNode? countersNode = xmlDoc.SelectSingleNode("/ns:TestRun/ns:ResultSummary/ns:Counters", nsManager);
+
+            XmlNode? resultSummaryNode = xmlDoc.SelectSingleNode("/ns:TestRun/ns:ResultSummary", nsManager);
+
+            XmlNode? startTimeNode = xmlDoc.SelectSingleNode("/ns:TestRun/ns:Times", nsManager);
+
+            XmlNodeList? unitTestResultNodes = xmlDoc.SelectNodes("/ns:TestRun/ns:Results/ns:UnitTestResult", nsManager);
+
+            if (countersNode == null || resultSummaryNode == null || startTimeNode == null)
+            {
+                throw new ArgumentException("Incorrect xml test file supplied, cannot find needed data");
+            }
+
+            string? passed = countersNode.Attributes?["passed"]?.Value;
+
+            string? failed = countersNode.Attributes?["failed"]?.Value;
+
+            string? outcome = resultSummaryNode.Attributes?["outcome"]?.Value;
+
+            string? startTimeString = startTimeNode.Attributes?["start"]?.Value;
+
+            string? finishTimeString = startTimeNode.Attributes?["finish"]?.Value;
```

Comment:
maybe it would be more readable to  move this to a separate method and return just an class with all these fields

Reply:
Separated to a separate method and created an according dto